### PR TITLE
Implemented cleanup on failure.

### PR DESCRIPTION
**Pull request by pykong**
This is almost ready to merge.

Two points:
- As the test run with `tmp/` as the destination, the test never gets a chance to actually remove. How can we address this?
- `shutil.rmtree` fail when called with a non-existing path, `ignore_errors` does not prevent this

Closes: #22

### CODE REVIEW

File: tests/test_cleanup.py [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+from .. import copier
+
+from .helpers import assert_file, render, PROJECT_TEMPLATE, DATA, filecmp
+
+
+@pytest.mark.slow
+def test_cleanup(dst):
+    with pytest.raises(CalledProcessError):
+        copier.copy("./tests/demo_cleanup", dst, quiet=True)
+    assert not (dst).exists()
```
Comment (pykong) [inline tests/test_cleanup.py]: The test will always find a non-existing path, as temporary dir gets removed.

File: copier/main.py [hunk 2]
```diff
@@ -137,6 +137,10 @@ def copy(
             skip=skip,
             quiet=quiet,
         )
+    except:  # noqa
+        print("Something went wrong. Removing destination dir.")
+        shutil.rmtree(dst_path, ignore_errors=True)
```
Comment (pykong) [inline copier/main.py]: Will raise when called with non-existent path.

**Comment by pykong** [comment]
It appears the error happens on the teardown of the test fixture.
The cleanup procedure I wrote successfully removes the temporary target, but this makes the teardown upset as it no longer has any dir to remove.

Explicitly ignoring errors resolves this problem:
https://github.com/jpscaletti/copier/blob/302dde9b17a8198e08fe8c8cf034ed226112f752/tests/conftest.py#L17

Changing to:

```py
request.addfinalizer(lambda: shutil.rmtree(dst, ignore_errors=True))
```

Is there any good reason not to do it?

**Comment by pykong** [comment]
@jpscaletti 

The tests are now green, except for `3.5`.
I do not have python 3.5 installed locally, as I am mainly using python 3.7.

Hence could you probably figure out what is going wrong?

**Comment by jpscaletti** [comment]
Is my fault, the tests were failing in master as well because of an extra comma.
However I probably wasn't clear enough at #22 .
I don't want that copier to auto-removes the destination folder if something fails.
I consider that a feature, not a bug.
But if you want to have it, it can be implemented in a python file that calls `copier`, since any  exception propagates up, but I don't think it should be in `copier`.

**Comment by jpscaletti** [comment]
Thanks ;)

**Pull request merged**